sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -148,80 +158,123 @@ def fit(self, X, y):
         # n_trees_per_iterations is n_classes in multiclass classification,
         # else 1.
         n_samples = X_binned_train.shape[0]
-        self._baseline_prediction = self.loss_.get_baseline_prediction(
-            y_train, self.n_trees_per_iteration_
-        )
-        raw_predictions = np.zeros(
-            shape=(self.n_trees_per_iteration_, n_samples),
-            dtype=self._baseline_prediction.dtype
-        )
-        raw_predictions += self._baseline_prediction
 
-        # initialize gradients and hessians (empty arrays).
-        # shape = (n_trees_per_iteration, n_samples).
-        gradients, hessians = self.loss_.init_gradients_and_hessians(
-            n_samples=n_samples,
-            prediction_dim=self.n_trees_per_iteration_
-        )
+        if not (self._has_been_fitted() and self.warm_start):
+            # Clear the state
```

Comment:
I would suggest "clear random state and score attributes" which is more descriptive


Also please add a short comment indicating both sections of the huge if/else since it's easy to lose track ;)   :

- "# first time calling fit, or no warm start"
- "# warm start: this is not the first time fit was called"